### PR TITLE
Fixed handling of PyTuple and PyList in converter

### PR DESCRIPTION
**Pull request by mailletf**
Lists and tuples were handled in the same if, which caused an exception in an actual prod plugin under certain scenarios. This fixes that by dealing with both cases seperately

### CODE REVIEW

File: plugins/lang/python/python_converters.cc [hunk 1]
```diff
@@ -230,7 +230,15 @@ construct_recur(PyObject * pyObj)
     //else if(PyDateTime_Check(pyObj)) {
     //    throw ML::Exception("do datetime!!");
     //}
-    else if(PyTuple_Check(pyObj) || PyList_Check(pyObj)) {
+    else if(PyTuple_Check(pyObj)) {
```
Comment (guyd) [inline plugins/lang/python/python_converters.cc]: The function could be turned into a function template to avoid the duplication.

**Comment by guyd** [comment]
+1 once the tests pass

**Comment by mailletf** [comment]
@guyd J'ai modifié le code pour suivre ta suggestion de template. Peux-tu rejeter un coup d’œil svp?

**Comment by guyd** [comment]
+1  I will a look at what is happening with the build.

**Pull request merged**